Doc/library/sqlite3.rst

```diff
@@ -47,84 +47,152 @@ This document includes four main sections:
-   cur.execute("INSERT INTO stocks VALUES ('2006-01-05','BUY','RHAT',100,35.14)")
+   cur.execute("CREATE TABLE Movie(Title, Year, Score)")
 
-   # Save (commit) the changes
-   con.commit()
+.. Ideally, we'd use sqlite_schema instead of sqlite_master below,
+   but earlier versions of SQLite does not recognise that variant.
 
-   # We can also close the connection if we are done with it.
-   # Just be sure any changes have been committed or they will be lost.
-   con.close()
+You can verify that the table has been created by querying
+the ever-present ``sqlite_master`` table.
+It should contain an entry for the ``Movie`` table definition.
+Execute the query by calling :meth:`cur.execute(...) <Cursor.execute>`,
+store the result in a variable ``res``,
+and call :meth:`res.fetchone() <Cursor.fetchone>` to fetch the first
+(and only) row from the query::
 
-The saved data is persistent: it can be reloaded in a subsequent session even
-after restarting the Python interpreter::
+   >>> res = cur.execute("SELECT name FROM sqlite_master where name='Movie'")
+   >>> res.fetchone()
+   ('Movie',)
 
-   import sqlite3
-   con = sqlite3.connect('example.db')
-   cur = con.cursor()
+The result is a one-item :class:`tuple`: one row, with one column.
+As expected, the query shows the table is now created.
+As an exercise, try querying ``sqlite_master`` for a bogus name ``"abc"``::
+
+   >>> res = cur.execute("SELECT name FROM sqlite_master where name='abc'")
+   >>> res.fetchone()
+   >>>
+
+As expected, the query returns an empty result.
+
+Now, add two rows of data supplied as SQL literals
+by executing an ``INSERT`` statement,
+once again by calling :meth:`cur.execute(...) <Cursor.execute>`::
 
-At this point, our database only contains one row::
+   cur.execute("""
+       INSERT INTO Movie VALUES
+           ('Monty Python and the Holy Grail', 1975, 8.2),
+           ('And Now for Something Completely Different', 1971, 7.5)
+   """)
 
-   >>> res = cur.execute('SELECT count(rowid) FROM stocks')
-   >>> print(res.fetchone())
-   (1,)
+The ``INSERT`` statement implicitly opens a transaction,
+which needs to be committed before changes are saved in the database
+(see :ref:`sqlite3-controlling-transactions` for details).
+Call :meth:`con.commit() <Connection.commit>` on the connection object
+to commit the transaction::
 
-The result is a one-item :class:`tuple`:
-one row, with one column.
-Now, let us insert three more rows of data,
-using :meth:`~Cursor.executemany`::
+   con.commit()
+
+Execute a query to verify that the data was inserted correctly.
+Use the now familiar :meth:`con.execute(...) <Cursor.execute>` to
+store the result in ``res``,
+and call :meth:`res.fetchall() <Cursor.fetchall>` to fetch all rows::
+
+   >>> res = cur.execute("SELECT Score FROM Movie")
+   >>> res.fetchall()
+   [(8.2,), (7.5,)]
+
+The result is a :class:`list` with two rows of data,
+each a one-item :class:`!tuple`,
+containing the value from the single column queried.
```

Comment:
```suggestion
The result is a :class:`list` of two :class:`!tuple`\ s (one per row),
each containing a single value (the ``Score`` from the query).
```

Reply:
IMO, that variation is less easy to understand, particularly due to the multiple parenthetical in the same sentence. Also, FYI you can just do ``:class:`!tuple`\s``; no need to add a space.

Reply:
I used a variant with less parentheses.